### DISCUSSION TITLE
Possibility to detail CC value Interfaces

### DISCUSSION THREAD

**Discussion by marcus-j-davies** [zwave-js/zwave-js#2976] [Q&A] [answered]
So,

The NR plugin is going through a huge change. i.e it will soon be able to call any CC method, provide durations without having to instantiate it's class, method names are now exact, enums can now be dynamaiclly referenced etc etc.

It's now much much closer to using Z-Wave JS in it's native form, but in node-red and using JSON alone.
Users can now view the list of CC's [Here](https://zwave-js.github.io/node-zwave-js/#/api/CCs/index), and use them in node-red. ( sorry, I felt like I was trying to pitch then)

All this has been made possible with the last few advancements - Awesome Work!

However, one thing bothers me.
there is no typed documentation on the various interfaces a CC might use, example:  the Association Methods, details the interface in documents, but it's very very rare (or not there at all), about each CC's interface if said CC uses one.

Given the magic you already conjure with docs, is there any way to also document the interfaces in each CC guide?

For CC's that use primitive types, this isn't  a problem of course, as its quite clear in docs, but when an interface is used, there is no documentation, one has to view the source code, to ascertain what the structure of the interface looks like. 

Which isn't a problem for the experienced user, but newbies into Home Automation, tend to struggle.

**Comment by AlCalzone** [comment, marked as answer]
This is certainly possible, but needs to be implemented. Can you open this as an issue, so it stays on my radar?

**Reply by marcus-j-davies** [reply]
Raised as enhancement.

https://github.com/zwave-js/node-zwave-js/issues/2978
